import "ImgDetections.proto";

syntax = "proto3";

package proto;

message SpatialImgDetections {
    int64 sequenceNum = 1;
    Timestamp ts = 2;
    Timestamp tsDevice = 3;
    repeated SpatialImgDetection detections = 4;
}

SpatialImgDetection {
    ImgDetection detection = 1;
    Point3f spatialCoordinates = 2;
    SpatialLocationCalculatorConfigData boundingBoxMapping = 3;
}

message Point3f {
    float x = 1; // default value: 0.0f
    float y = 2; // default value: 0.0f
    float z = 3; // default value: 0.0f
}

message SpatialLocationCalculatorConfigData {
    int32_t AUTO = 1; // default value: -1
    Rect roi = 2;
    SpatialLocationCalculatorAlgorithm calculationAlgorithm = 3;
    int32 stepSize = 4; // default value: AUTO
}


message Rect {
    float x = 1; // default value: 0.0f
    float y = 2; // default value: 0.0f
    float width = 3; // default value: 0.0f
    float height = 4; // default value: 0.0f
}

message SpatialLocationCalculatorAlgorithm {
    uint32 lowerThreshold = 1; // default value: 0
	uint32 upperThreshold = 2; // default value: 65535
}
